src/evaluators/interpolation/PHAL_DOFCellToSide_Def.hpp

```diff
@@ -117,24 +140,230 @@ postRegistrationSetup(typename Traits::SetupData d,
 
   val_side.dimensions(dims);
 
+  dimsView = Kokkos::DynRankView<int, PHX::Device>("dimsView", dims.size());
+  for (int i = 0; i < dims.size(); ++i)
+    dimsView(i) = dims[i];
+
   d.fill_field_dependencies(this->dependentFields(),this->evaluatedFields());
   if (d.memoizer_active()) memoizer.enable_memoizer();
 }
 
+// *********************************************************************
+// Kokkos functor
+template<typename EvalT, typename Traits, typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+void DOFCellToSideBase<EvalT, Traits, ScalarT>::
+operator() (const DOFCellToSide_CellScalar_Tag& tag, const int& sideSet_idx) const {
+  
+  // Get the local data of side and cell
+  const int cell = sideSet.elem_LID(sideSet_idx);
+  const int side = sideSet.side_local_id(sideSet_idx);
+
+  val_side(cell,side) = val_cell(cell);
+
+}
+
+template<typename EvalT, typename Traits, typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+void DOFCellToSideBase<EvalT, Traits, ScalarT>::
+operator() (const DOFCellToSide_CellVector_Tag& tag, const int& sideSet_idx) const {
+
+  // Get the local data of side and cell
+  const int cell = sideSet.elem_LID(sideSet_idx);
+  const int side = sideSet.side_local_id(sideSet_idx);
+
+  for (int i=0; i<dimsView(2); ++i) {
+    val_side(cell,side,i) = val_cell(cell,i);
+  }
+
+}
+
+template<typename EvalT, typename Traits, typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+void DOFCellToSideBase<EvalT, Traits, ScalarT>::
+operator() (const DOFCellToSide_CellTensor_Tag& tag, const int& sideSet_idx) const {
+
+  // Get the local data of side and cell
+  const int cell = sideSet.elem_LID(sideSet_idx);
+  const int side = sideSet.side_local_id(sideSet_idx);
+
+  for (int i=0; i<dimsView(2); ++i) {
+    for (int j=0; j<dimsView(3); ++j) {
+      val_side(cell,side,i,j) = val_cell(cell,i,j);
+    }
+  }
+
+}
+
+template<typename EvalT, typename Traits, typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+void DOFCellToSideBase<EvalT, Traits, ScalarT>::
+operator() (const DOFCellToSide_NodeScalar_Tag& tag, const int& sideSet_idx) const {
+
+  // Get the local data of side and cell
+  const int cell = sideSet.elem_LID(sideSet_idx);
+  const int side = sideSet.side_local_id(sideSet_idx);
+
+  for (int node=0; node<dimsView(2); ++node) {
+      val_side(cell,side,node) = val_cell(cell,sideNodes(side,node));
```

Comment:
Would subviewing the input view as (cell,side,ALL) and the output view at (cell,ALL) improve performance by reducing indexing time?

Reply:
I think it would but it might also increase code complexity.